pkgs/stdenv/adapters.nix

```diff
@@ -261,25 +261,27 @@ rec {
     in
     overrideCC targetStdenv cc;
 
-  useMoldLinker = stdenv: let
-    bintools = stdenv.cc.bintools.override {
-      extraBuildCommands = ''
-        wrap ${stdenv.cc.bintools.targetPrefix}ld.mold ${../build-support/bintools-wrapper/ld-wrapper.sh} ${pkgs.mold}/bin/ld.mold
-        wrap ${stdenv.cc.bintools.targetPrefix}ld ${../build-support/bintools-wrapper/ld-wrapper.sh} ${pkgs.mold}/bin/ld.mold
-      '';
-    };
-  in stdenv.override (old: {
-    allowedRequisites = null;
-    cc = stdenv.cc.override { inherit bintools; };
-    # gcc >12.1.0 supports '-fuse-ld=mold'
-    # the wrap ld above in bintools supports gcc <12.1.0 and shouldn't harm >12.1.0
-    # https://github.com/rui314/mold#how-to-use
-    } // lib.optionalAttrs (stdenv.cc.isClang || (stdenv.cc.isGNU && lib.versionAtLeast stdenv.cc.version "12")) {
-    mkDerivationFromStdenv = extendMkDerivationArgs old (args: {
-      NIX_CFLAGS_LINK = toString (args.NIX_CFLAGS_LINK or "") + " -fuse-ld=mold";
-    });
-  });
-
+  useMoldLinker = stdenv:
+    if stdenv.targetPlatform.isDarwin
+    then throw "Mold can't be used to emit Mach-O (Darwin) binaries"
+    else let
+      bintools = stdenv.cc.bintools.override {
+        extraBuildCommands = ''
+          wrap ${stdenv.cc.bintools.targetPrefix}ld.mold ${../build-support/bintools-wrapper/ld-wrapper.sh} ${pkgs.mold}/bin/ld.mold
+          wrap ${stdenv.cc.bintools.targetPrefix}ld ${../build-support/bintools-wrapper/ld-wrapper.sh} ${pkgs.mold}/bin/ld.mold
+        '';
+      };
+    in stdenv.override (old: {
+      allowedRequisites = null;
+      cc = stdenv.cc.override { inherit bintools; };
+      # gcc >12.1.0 supports '-fuse-ld=mold'
+      # the wrap ld above in bintools supports gcc <12.1.0 and shouldn't harm >12.1.0
+      # https://github.com/rui314/mold#how-to-use
+      } // lib.optionalAttrs (stdenv.cc.isClang || (stdenv.cc.isGNU && lib.versionAtLeast stdenv.cc.version "12")) {
+        mkDerivationFromStdenv = extendMkDerivationArgs old (args: {
+          NIX_CFLAGS_LINK = toString (args.NIX_CFLAGS_LINK or "") + " -fuse-ld=mold";
+        });
+      });
```

Comment:
Let's `nixfmt` this bit, since it's changing so much.

```suggestion
  useMoldLinker =
    stdenv:
    if stdenv.targetPlatform.isDarwin then
      throw "Mold can't be used to emit Mach-O (Darwin) binaries"
    else
      let
        bintools = stdenv.cc.bintools.override {
          extraBuildCommands = ''
            wrap ${stdenv.cc.bintools.targetPrefix}ld.mold ${../build-support/bintools-wrapper/ld-wrapper.sh} ${pkgs.mold}/bin/ld.mold
            wrap ${stdenv.cc.bintools.targetPrefix}ld ${../build-support/bintools-wrapper/ld-wrapper.sh} ${pkgs.mold}/bin/ld.mold
          '';
        };
      in
      stdenv.override (
        old:
        {
          allowedRequisites = null;
          cc = stdenv.cc.override { inherit bintools; };
          # gcc >12.1.0 supports '-fuse-ld=mold'
          # the wrap ld above in bintools supports gcc <12.1.0 and shouldn't harm >12.1.0
          # https://github.com/rui314/mold#how-to-use
        }
        //
          lib.optionalAttrs
            (stdenv.cc.isClang || (stdenv.cc.isGNU && lib.versionAtLeast stdenv.cc.version "12"))
            {
              mkDerivationFromStdenv = extendMkDerivationArgs old (args: {
                NIX_CFLAGS_LINK = toString (args.NIX_CFLAGS_LINK or "") + " -fuse-ld=mold";
              });
            }
      );
```

Reply:
All formatting commits have to be added to `.git-blame-ignore-revs` so this would have to be in a separate commit, to preserve the performance of git it would be better to format the whole file in one commit.

Reply:
I buy that, but this PR's content over most of these lines IS a formatting change -- so making it be the nixfmt version, not a hand rolled version, makes sense to me.

Reply:
> I buy that, but this PR's content over most of these lines IS a formatting change -- so making it be the nixfmt version, not a hand rolled version, makes sense to me.

It's only whitespace changes which `git blame` can ignore with `-w`

Reply:
OK, I buy that. I'm using the default GitHub view, which doesn't ignore whitespace.